pkg/gadgettracermanager/gadgettracermanager_test.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gadgettracermanager
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
+)
+
+func TestSelector(t *testing.T) {
+	table := []struct {
+		description string
+		match       bool
+		selector    *pb.ContainerSelector
+		container   *pb.ContainerDefinition
+	}{
+		{
+			description: "Selector without filter",
+			match:       true,
+			selector:    &pb.ContainerSelector{},
+			container: &pb.ContainerDefinition{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+			},
+		},
+		{
+			description: "Selector without all filters",
+			match:       true,
+			selector: &pb.ContainerSelector{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+				Labels: []*pb.Label{
+					&pb.Label{Key: "key1", Value: "value1"},
+					&pb.Label{Key: "key2", Value: "value2"},
+				},
+			},
+			container: &pb.ContainerDefinition{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+				Labels: []*pb.Label{
+					&pb.Label{Key: "key1", Value: "value1"},
+					&pb.Label{Key: "key2", Value: "value2"},
```

Comment:
We can add a third label and it still will match, right? (just to check that the matching algorithm matches a subset of labels)